src/monster.cpp

```diff
@@ -3070,11 +3092,14 @@ void monster::hear_sound( const tripoint &source, const int vol, const int dist,
     if( morale >= 0 && anger >= 10 ) {
         // TODO: Add a proper check for fleeing attitude
         // but cache it nicely, because this part is called a lot
-        wander_to( tripoint( target, source.z ), wander_turns );
+        wander_to( target, wander_turns );
     } else if( morale < 0 ) {
         // Monsters afraid of sound should not go towards sound
-        wander_to( -target + tripoint( 2 * posx(), 2 * posy(), 2 * posz() - source.z ),
-                   wander_turns );
+        // TODO: make the destination scale with the sound and handle
+        // the case when (x,y) is the same by picking a random direction
+        tripoint_abs_ms away = get_location() - target + get_location();
```

Comment:
Took me a while to figure out why this strange computation.  Might warrant a comment.
```suggestion
        // Move towards a point on the opposite side of us from the target
        tripoint_abs_ms away = get_location() + ( get_location() - target );
```

Reply:
Done.